bot/exts/info/information.py

```diff
@@ -42,15 +43,32 @@ def get_channel_type_counts(guild: Guild) -> DefaultDict[str, int]:
         return channel_counter
 
     @staticmethod
-    def get_member_counts(guild: Guild) -> Dict[str, int]:
+    def join_role_stats(role_ids: list[int], name: str, guild: Guild) -> dict[str, int]:
+        """Return a dictionary with the number of `members` of each role given, and the `name` for this joined group."""
+        members = []
+        for role_id in role_ids:
+            if (role := guild.get_role(role_id)) is None:
+                raise NonExistentRoleError(role_id)
+            else:
+                members += role.members
+        return {name: len(set(members))}
+
+    @staticmethod
+    def get_member_counts(guild: Guild) -> dict[str, int]:
         """Return the total number of members for certain roles in `guild`."""
-        roles = (
-            guild.get_role(role_id) for role_id in (
-                constants.Roles.helpers, constants.Roles.mod_team, constants.Roles.admins,
-                constants.Roles.owners, constants.Roles.contributors,
-            )
+        role_ids = [constants.Roles.helpers, constants.Roles.mod_team, constants.Roles.admins,
+                    constants.Roles.owners, constants.Roles.contributors]
+        roles = []
+        for role_id in role_ids:
+            if (role := guild.get_role(role_id)) is None:
+                raise NonExistentRoleError(role_id)
+            else:
+                roles.append(role)
```

Comment:
There's some unnecessary code-duplication here.

Reply:
I was reminded that if there is a terrible misconfiguration and those essential roles, such as helpers, or owners do not exist this bit of code would fail, so I added an additional check. In case of the other bit of code (`join_role_stats`) the roles that need to be joined might also not exist so I had to add another check)
I'm not sure how to get around this, the roles that needs to be joined must be handled differently than the ones that don't.